### PR TITLE
Update the WASM filter user documentation under example/

### PR DESCRIPTION
**Pull request by incfly**
For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description:

- Explained the necessary setup 
- Update outdated lua filter part .

### CODE REVIEW

File: WASM.md [hunk 1]
```diff
@@ -21,3 +21,26 @@ proof of concept because:
...
+
+```shell
+bazel build  --define wasm=wavm  //source/exe:envoy-static
+```
+
+The available VM name options are
+
+1. wavm
+1. v8
+1. null? (Avaialbe or not?)
```
Comment (PiotrSikora) [inline WASM.md]: The options are:
- `wasm=enabled` - both: V8 & WAVM,
- `wasm=v8` - only V8,
- `wasm=wavm` - only WAVM.

Null sandbox is always available.

File: WASM.md [hunk 2]
```diff
@@ -21,3 +21,26 @@ proof of concept because:
...
+
+
+To build Envoy via Docker image, pass `--define wasm=<vm-name>` as `BAZEL_EXTRA_BUILD_OPTIONS`
+
+```shell
+BAZEL_EXTRA_BUILD_OPTIONS='--define wasm=wavm' ./ci/run_envoy_docker.sh './ci/do_ci.sh bazel.dev'
```
Comment (PiotrSikora) [inline WASM.md]: You need to set `BAZEL_EXTRA_BUILD_OPTIONS` inside Docker's shell, e.g.
```
./ci/run_envoy_docker.sh 'BAZEL_EXTRA_BUILD_OPTIONS=--define=wasm=v8 ./ci/do_ci.sh bazel.dev'
```

File: examples/wasm/README.md [hunk 3]
```diff
@@ -1,86 +1,76 @@
-In this example, we show how a Lua filter can be used with the Envoy
-proxy. The Envoy proxy [configuration](./envoy.yaml) includes a lua
-filter that contains two functions namely
-`envoy_on_request(request_handle)` and
-`envoy_on_response(response_handle)` as documented
-[here](https://www.envoyproxy.io/docs/envoy/latest/configuration/http_filters/lua_filter).
+# Envoy Webassembly Filter
```
Comment (PiotrSikora) [inline examples/wasm/README.md]: Nit: it's "WebAssembly".

File: examples/wasm/README.md [hunk 4]
```diff
@@ -1,86 +1,76 @@
...
+example body
 * Connection #0 to host localhost left intact
-}
 ```
 
-# Build WASM Binary
-## prerequisite
-1. [Download and install `emscripten`](https://emscripten.org/docs/getting_started/downloads.html#installation-instructions)
-2. Activate emsdk archived version 1.38.25 (Note: by 02/11/2019 the latest build is broken for envoy-wasm)
+## Build WASM Binary
```
Comment (PiotrSikora) [inline examples/wasm/README.md]: Nit: "extension" or "module".

**Comment by PiotrSikora** [comment]
cc @duderino since he's been playing and running into similar issues with WASM lately.

File: examples/wasm/README.md (hunk 4 above)
Comment (incfly) [reply examples/wasm/README.md]: Done.

File: examples/wasm/README.md (hunk 3 above)
Comment (incfly) [reply examples/wasm/README.md]: Done.

File: WASM.md (hunk 2 above)
Comment (incfly) [reply WASM.md]: Yeah, I figured it out and make a PR in upstream doc as well.

File: WASM.md (hunk 1 above)
Comment (incfly) [reply WASM.md]: Got it. Changed.

**Comment by duderino** [comment]
I think https://github.com/envoyproxy/envoy-wasm/pull/81 has more stuff in it than this one.  Should we merge https://github.com/envoyproxy/envoy-wasm/pull/81 then rebase this one?  @incfly @jplevyak @PiotrSikora

**Comment by incfly** [comment]
#81 may have more stuff in `WASM.md` for WASM core runtime developers like us.

This PR has `example/wasm/` related doc, which is wasm filter developer facing. I'm fine with either one, merging #81 then I can move forward this PR sounds good to me.

**Comment by duderino** [comment]
Good point @incfly we want both

**Comment by PiotrSikora** [comment]
We want both. Let's rebase this on top of #81 once it's merged.

**Comment by PiotrSikora** [comment]
@incfly could you rebase this, now that #81 is merged? Thanks!

**Comment by incfly** [comment]
Done. @PiotrSikora

**Comment by PiotrSikora** [comment]
@jplevyak can you take a look?

File: examples/wasm/README.md [hunk 5]
```diff
@@ -1,86 +1,76 @@
...
-```shell
-envoy/examples/wasm$ source <path/to/emsdk>/emsdk_env.sh  # Only needed if in another shell session
-envoy/examples/wasm$ touch envoy_filter_http_wasm_example.cc && make
-```
+### prerequisite
+
+1. Download and install [`protobuf`](https://github.com/protocolbuffers/protobuf/blob/master/src/README.md).
+1. Download and install [`WAVM`](https://github.com/WAVM/WAVM).
+1. Download and install [`emscripten`](https://emscripten.org/docs/getting_started/downloads.html#installation-instructions)
+1. Activate emsdk archived version 1.38.25 (Note: by 02/11/2019 the latest build is broken for envoy-wasm)
```
Comment (jplevyak) [inline examples/wasm/README.md]: The latest (1.38.39) works with the latest master.

File: examples/wasm/README.md [hunk 6]
```diff
@@ -1,86 +1,76 @@
...
+
+    ```shell
+    # in emsdk directory
+    ./emsdk install sdk-1.38.25-64bit
```
Comment (jplevyak) [inline examples/wasm/README.md]: perhaps we can say 1.38.39

File: examples/wasm/README.md (hunk 5 above)
Comment (incfly) [reply examples/wasm/README.md]: done. i reworded to update the latest known working version.

File: examples/wasm/README.md (hunk 6 above)
Comment (incfly) [reply examples/wasm/README.md]: s/.25/.39/g

File: examples/wasm/README.md [hunk 7]
```diff
@@ -1,86 +1,76 @@
...
+    ./emsdk activate sdk-1.38.25-64bit
+    source emsdk_env.sh
+    # verify, should output 1.38.25
```
Comment (jplevyak) [inline examples/wasm/README.md]: 38.39

File: examples/wasm/README.md [hunk 8]
```diff
@@ -1,86 +1,76 @@
...
-1. `docker-compose build`
-2. `docker-compose up`
-3. `curl -v localhost:8000`
 
-## Sample Output:
+## Quick Start
+
+1. `docker-compose build`
+1. `docker-compose up`
+1. `curl -v localhost:18000`
```
Comment (PiotrSikora) [inline examples/wasm/README.md]: Nit: why all list elements start with `1.` in this file? I know it's generated properly on GitHub, but it's a bit annoying when looking at the raw file.

**Comment by incfly** [comment]
I may adopt this style from g3doc internally.
https://github.com/google/styleguide/blob/gh-pages/docguide/style.md#use-lazy-numbering-for-long-lists

I'm fine with changing 1. 2. 3 as well.

**Pull request merged**